src/main/java/org/apache/accumulo/access/AccessEvaluator.java

```diff
@@ -117,15 +131,31 @@ interface AuthorizationsBuilder {
      *
      *
      */
-    ExecutionBuilder authorizations(Collection<Authorizations> authorizations);
+    OptionalBuilder authorizations(Collection<Authorizations> authorizations);
 
-    ExecutionBuilder authorizations(String... authorizations);
+    /**
+     * Allows specifying a single set of authorizations.
+     */
+    OptionalBuilder authorizations(String... authorizations);
 
-    ExecutionBuilder authorizations(Authorizer authorizer);
+    /**
+     * Allows specifying an authorizer that is analogous to a single set of authorization.
+     */
+    OptionalBuilder authorizations(Authorizer authorizer);
   }
 
-  interface ExecutionBuilder extends FinalBuilder {
-    ExecutionBuilder cacheSize(int cacheSize);
+  interface OptionalBuilder extends FinalBuilder {
+
+    /**
+     * When set to a value greater than zero, the result of evaluating expressions will be
+     * remembered and if the same expression is seen it again the remembered result will be used
+     * instead of reevaluating it. If this method is not called on the builder then no caching is
+     * done. When the same expressions can occur repeatedly caching can greatly increase
+     * performance.
+     *
+     * @param cacheSize the number of expressions evaluations to remember in an LRU cache.
+     */
+    OptionalBuilder cacheSize(int cacheSize);
```

Comment:
Instead of having a built-in internal cache with only an option for a fixed size, I wonder if it'd be better to just have a pluggable cache, so users can provide whatever cache implementation they want, rather than only control the size.

Reply:
Maybe if it would be best to just drop this functionality. It would be really easy for someone to wrap the evaluator with their own cache.  Could just recommend caching in the class level javadocs.

Reply:
Opened #17